### ISSUE TITLE
Missing provenance warning not emitted for -Wint-conversion

### ISSUE THREAD

**Issue by arichardson** [CTSRD-CHERI/llvm-project#487]
```c
void *explicit(long long int x) {
  return (void *)x;
}

void *implicit(long long int x) {
  return x;
}
```

The first case triggers
```
<source>:4:10: warning: cast from provenance-free integer type to pointer type will give pointer that can not be dereferenced [-Wcheri-capability-misuse]
  return (void *)x;
         ^
```
but the second only results in
```
<source>:8:10: warning: incompatible integer to pointer conversion returning 'long long' from a function with result type 'void *' [-Wint-conversion]
  return x;
```
and does not emit any warning if you compile with -Wno-int-conversion.
